src/AzureAppConfigurationImpl.ts

```diff
@@ -29,20 +36,41 @@ export class AzureAppConfigurationImpl extends Map<string, unknown> implements A
         super();
         // Enable request tracing if not opt-out
         this.requestTracingEnabled = requestTracingEnabled();
-        if (this.requestTracingEnabled) {
-            this.enableRequestTracing();
-        }
 
         if (options?.trimKeyPrefixes) {
             this.sortedTrimKeyPrefixes = [...options.trimKeyPrefixes].sort((a, b) => b.localeCompare(a));
         }
+
+        if (options?.refreshOptions) {
+            this.onRefreshListeners = new LinkedList();
+            this.refreshIntervalInMs = DefaultRefreshIntervalInMs;
+
+            const refreshIntervalInMs = this.options?.refreshOptions?.refreshIntervalInMs;
+            if (refreshIntervalInMs !== undefined) {
+                if (refreshIntervalInMs < MinimumRefreshIntervalInMs) {
+                    throw new Error(`The refresh interval time cannot be less than ${MinimumRefreshIntervalInMs} milliseconds.`);
+                } else {
+                    this.refreshIntervalInMs = refreshIntervalInMs;
+                }
+            }
+
+            this.sentinels = options.refreshOptions.watchedSettings?.map(setting => {

```

Comment:
why it's different between `options.refreshOptions.watchedSettings` and `this.options?.refreshOptions?.refreshIntervalInMs`? where the latter calls `this` and uses nullable`?.`

Reply:
1. `options` is equivalent to `this.options` in the constructor. It's a grammar sugar.
```ts
// simplified
class Foo {
  constructor(private bar: string) {}
}

// equivalent
class Foo {
  private bar: string;
  constructor(bar: string) {
    this.bar = bar;
  }
}
```

Reply:
2. about `?.`, it's another convenient grammar sugar. 

```ts
// short
a?.b 
// equivalent
a ? a.b : undefined
```

Without `?` it's possible to have NPE like access some property of undefined. The chained `?.` will be super convenient if you want to access something deep inside optional properties.

Reply:
I mean what's difference between `watchedSettings` and `refreshIntervalInMs`? one needs null check, another doesn't

Reply:
Good catch.  They are in the block that `options.refreshOptions` is not null. But `this.options.refreshOptions` can be null (impossible, only from perspective of TypeScript type deduction), and that's why I added the null check.

I shall change `this.options?.refreshOptions?.refreshIntervalInMs` to `options.refreshOptions.refreshIntervalInMs`.

And based on the discussion we'll add a property `enabled` in `refreshOptions`, explictly indicating whether to enable the feature. I'll re-organize and simplify the impl later.

Reply:
fixed.